File: lib/models/menu_item.dart
class MenuItem {
  final String id;
  final String name;
  final String description;
  final double price;
  final bool isVeg;
  final String imageUrl;
  final String categoryId;

  MenuItem({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.isVeg,
    required this.imageUrl,
    required this.categoryId,
  });

  factory MenuItem.fromJson(Map<String, dynamic> json) {
    return MenuItem(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] is num) ? (json['price'] as num).toDouble() : 0.0,
      isVeg: json['isVeg'] ?? false,
      imageUrl: json['imageUrl'] ?? '',
      categoryId: json['categoryId'] ?? '',
    );
  }
}

File: server/models/menuItem.js
const mongoose = require('mongoose');

const menuItemSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    required: true,
    trim: true
  },
  price: {
    type: Number,
    required: true,
    min: 0
  },
  isVeg: {
    type: Boolean,
    default: false
  },
  imageUrl: {
    type: String,
    required: true
  },
  categoryId: {
    type: String,
    required: true
  },
  restaurantId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Restaurant',
    required: true
  }
}, { timestamps: true });

const MenuItem = mongoose.model('MenuItem', menuItemSchema);

module.exports = MenuItem;
